### PR TITLE
DSI 0.4.0 and Saved Query Exports

### LINKED ISSUE #8892: [CT-3255] [implementation] interface changes to dbt-core saved_query spec to add export
### Housekeeping

- [X] I am a maintainer of dbt-core

### Short description

Add exports to saved_query

```
saved_queries:
  - name: my_query
    description:
    query:
      metrics:
        - 
      group_bys:
        - TimeDimension()
        - ...
      where:
        - 
    exports:
      - name: my_query 
        config:
          export_as: table # options: table, view - FUTURE: incremental_table, window_table, file
          schema: my_schema  # [optional - DEFAULT to deployment schema]
          alias: some_table_name # [optional - DEFAULT to export name]
```

### Acceptance criteria

- add exports to saved_query

### Impact to Other Teams

SL

### Will backports be required?

1.7 latest

### Context

_No response_

### PR DESCRIPTION
**Pull request by QMalcolm**
resolves #8892 

### Problem
dbt-semantic-interfaces recently released 0.4.0. We want to support this for 1.7.0 of dbt-core. dbt-semantic-interfaces 0.4.0 has a few changes from 0.3.0

- `where`, `group_by`, and `metrics` are now nested under the `query_params` property on a `SavedQuery`
- The `SavedQuery` now has a concept of `exports`

### Solution

Begin supporting the changes. 

### Other things to flag
In 0.4.0 the names of metrics begin being validated by dbt-semantic-interfaces. This was always supposed to be happening, but wasn't 😬 This might break for some people. To be clear it was likely breaking in the semantic layer anyways. This just ensures that the issue is caught at parse time rather than query time.

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [X] I have run this code in development and it appears to resolve the stated issue  
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [X] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [X] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

File: core/dbt/contracts/graph/nodes.py [hunk 1]
```diff
@@ -1783,6 +1768,17 @@ def same_config(self, old: "SavedQuery") -> bool:
     def same_group(self, old: "SavedQuery") -> bool:
         return self.group == old.group
 
+    def same_exports(self, old: "SavedQuery") -> bool:
+        if len(self.exports) != len(old.exports):
```
Comment (ChenyuLInx) [inline core/dbt/contracts/graph/nodes.py:1772]: Is this check here sufficient? seems like we might be updating exports but didn't realizing it?

**Review by ChenyuLInx — APPROVED** [review]
LGTM, where are we defining using DSI 0.4.x?

[My PR](https://github.com/dbt-labs/dbt-core/pull/8937) probably needed to be updated if this goes in first, or the other way around

**Comment by QMalcolm** [comment]
Rebasing off of main to resolve conflicts

**Comment by QMalcolm** [comment]
> LGTM, where are we defining using DSI 0.4.x?
> 
> 
> 
> [My PR](https://github.com/dbt-labs/dbt-core/pull/8937) probably needed to be updated if this goes in first, or the other way around

We define the version in 'core/setup.py'

**Pull request merged**